src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -27,17 +34,62 @@ const iconClassnames = {
   error: `${iconBaseClassName}--error`,
 };
 
-export interface CheckboxProps extends CompCheckboxProps, TokensProp {}
+type CheckboxStatus = Exclude<InputStatus, 'success'>;
 
-const StyledCheckbox = styled(
-  ({ tokens, ...passProps }: CheckboxProps & InternalTokensProp) => (
-    <CompCheckbox {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
-`;
+export interface CheckboxProps extends TokensProp {
+  /** Controlled checked-state - user actions use onClick to change  */
+  checked?: boolean;
+  /** Default status of Checkbox when not using controlled 'checked' state
+   * @default false
+   */
+  defaultChecked?: boolean;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Disable Checkbox. Value won't be included when submitting */
+  disabled?: boolean;
+  /** Event handler to execute when clicked */
+  onClick?: ({ checkboxState }: { checkboxState: boolean }) => void;
+  /**
+   * Label element content
+   */
+  children?: ReactNode;
+  /**
+   * Variant of the Checkbox
+   * @default small
+   */
+  variant?: 'small' | 'large';
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: CheckboxStatus;
+  /**
+   * Status text to be displayed in the status text element. Will not be displayed when element is disabled.
+   */
+  statusText?: string;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /**
+   * aria-label for the HTML input-element,
+   * alternatively you can define aria-labelledby with label-element id
+   */
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  'aria-describedby'?: string;
```

Comment:
`aria-describedby` was added to interface as it was required to work as the Checkbox is not extending any interface that brings it.

For example, TextInput extends`HtmlInputProps`

https://github.com/vrk-kpa/suomifi-ui-components/blob/dccf99d269c4e9f1e9c0df1901aac8fa32288fce/src/core/Form/TextInput/TextInput.tsx#L33-L35

So, should **Checkbox** also extend it or was there reason why it is not already doing this?

Reply:
In my opinion components should extend the interface of the element where passprops are given. E.g. if everything is wrapped inside a div and passProps go there, it should extend HtmlDiv props (Or low level API if no resets are used). Or if passprops go to input element, that is what should be included in the API as well.

This is a bit difficult topic over all, but I think a good rule of thumb is to extend the thing the component is mostly about, input, text (can be div etc), button and so on.

Reply:
And in this case if Checkbox would extend HtmlInputProps or similar, we could provide containerProps: HtmlDivProps if necessary. In some cases it's also possible that extending would provide too much undesired or conflicting functionality and can be done the other way around by picking only necessary/needed props. So has to be considered case by case.

Reply:
We discussed about this and we leave it now as it is.

Reply:
As discussed elsewhere, this is ok for now. Using HtmlInputProps as passProps would introduce quite a few conflicts and would unnecessarily pollute the API without providing any mandatory functionality in addition to what is now available.